shim.c

```diff
@@ -1096,7 +1096,7 @@ EFI_STATUS start_image(EFI_HANDLE image_handle, CHAR16 *ImagePath)
 	void *sourcebuffer = NULL;
 	UINT64 sourcesize = 0;
 	void *data = NULL;
-	int datasize;
+	int datasize = 0;
```

Comment:
Should this be a separate commit?

Reply:
This still falls under category "fix compilation for older gcc". This is an issue for 4.8.5 and older(?) gcc, new enough will be fine with just int datasize;

Reply:
Actually this is fine, it is always set before reading.